indra/explanation/model_checker/model_checker.py

```diff
@@ -0,0 +1,490 @@
+    paths : list[list[tuple[str, int]]]
+        A list of paths obtained from path finding. Each path is a list of
+        tuples (which are edges in the path), with the first element of the
+        tuple the name of a rule, and the second element its polarity in the
+        path.
+    """
+    def __init__(self, path_found, result_code, max_paths, max_path_length):
+        self.path_found = path_found
+        self.result_code = result_code
+        self.max_paths = max_paths
+        self.max_path_length = max_path_length
+        self.path_metrics = []
+        self.paths = []
+
+    def add_path(self, path):
+        self.paths.append(path)
+
+    def add_metric(self, path_metric):
+        self.path_metrics.append(path_metric)
+
+    @python_2_unicode_compatible
+    def __str__(self):
+        summary = textwrap.dedent("""
+            PathResult:
+                path_found: {path_found}
+                result_code: {result_code}
+                path_metrics: {path_metrics}
+                paths: {paths}
+                max_paths: {max_paths}
+                max_path_length: {max_path_length}""")
+        ws = '\n        '
+        # String representation of path metrics
+        if not self.path_metrics:
+            pm_str = str(self.path_metrics)
+        else:
+            pm_str = ws + ws.join(['%d: %s' % (pm_ix, pm) for pm_ix, pm in
+                                   enumerate(self.path_metrics)])
+
+        def format_path(path, num_spaces=11):
+            path_ws = '\n' + (' ' * num_spaces)
+            return path_ws.join([str(p) for p in path])
+
+        # String representation of paths
+        if not self.paths:
+            path_str = str(self.paths)
+        else:
+            path_str = ws + ws.join(['%d: %s' % (p_ix, format_path(p))
+                                     for p_ix, p in enumerate(self.paths)])
+
+        return summary.format(path_found=self.path_found,
+                              result_code=self.result_code,
+                              max_paths=self.max_paths,
+                              max_path_length=self.max_path_length,
+                              path_metrics=pm_str, paths=path_str)
+
+    def __repr__(self):
+        return str(self)
+
+
+class ModelChecker(object):
+    """The parent class of all ModelCheckers.
+
+    Parameters
+    ----------
+    model : pysb.Model or indra.assemblers.indranet.IndraNet or PyBEL.Model
+        Depending on the ModelChecker class, can be different type.
+    statements : Optional[list[indra.statements.Statement]]
+        A list of INDRA Statements to check the model against.
+    do_sampling : bool
+        Whether to use breadth-first search or weighted sampling to
+        generate paths. Default is False (breadth-first search).
+    seed : int
+        Random seed for sampling (optional, default is None).
+
+    Attributes
+    ----------
+    graph : nx.Digraph
+        A DiGraph with signed nodes to find paths in.
+    """
```

Comment:
In the case of simple DiGraphs, it's not a requirement that the nodes be signed, correct?